### ISSUE TITLE
Unknown type: UnionType

### ISSUE THREAD

**Issue by danzel** [lukeautry/tsoa#5]
Make a controller with a return result that includes a string union
```
interface Result {
    value: 'success' | 'failure';
}
```
Then try generate routes with tsoa.
You get the error: `Unknown type: UnionType`

**Comment by lukeautry** [comment]
Good find, should be able to look at this today.

**Comment by lukeautry** [comment]
Started looking into the implementation for this, but I think there's a challenging limitation of Swagger/OpenAPI we'll have to figure out.

[Swagger spec deliberately disallows multiple response types ("union types")](https://github.com/OAI/OpenAPI-Specification/issues/270)

So, the question becomes: what should happen if a referenced model in your API code has a union type property, or the return type of an endpoint itself is a union type?

We can:

- Throw an error during generation (this is what happens now, and is obviously not great)
- We could somehow try to find a merged/common type between union types, or merge the type info into a combined type (this seems super confusing to me)
- We can find some way to essentially represent a union type as dynamic, i.e. no generated swagger definition (not super useful to a consumer, but at least you'd know that that an endpoint returning type "Result" has _some_ value by key 'value'. 

I think from an API design standpoint, an enum makes a lot more sense here than the union type string values. That said, there's no support in this library for enums at the moment, although I plan to implement that soon.

**Comment by lukeautry** [comment]
@danzel After further reflection I've decided on option #3 above - see https://github.com/lukeautry/tsoa/commit/305dd3221078fa705c22c8f591e867c690e13939

TLDR: union types will be set to type 'object' in swagger generation, which you can think of as 'any' or 'dynamic' type since Swagger/OpenAPI doesn't offer a way to represent this.

Let me know if this doesn't quite meet your expectations or if you have some additional thoughts to share.

**Comment by danzel** [comment]
You are right that I wanted to use an enum, hadn't actually tried an enum type though.

I'll make an issue for adding that, thanks!

**Comment by danzel** [comment]
Oh you beat me to it :)

**Comment by krueglt** [comment]
Actually swagger has a way of expressing "any". It is omitting the "type" property.
So in a schema instead of 
```
"values": {
          "type": "object",
           "properties": {}
          "description": "descr"
        }
```

you can have just:
```
"values": {
          "description": "descr"
        }
```

Having object as type has another meaning, since it says the type will be an object - and not a string, number, boolean..
I suggest removing the "type" property completely for any, and also for union types. Though this definitely requires more testing, i am not sure if omitting the type is allowed in all cases.

regards,
Thomas
